pkg/adaptor/adaptor.go

```diff
@@ -24,15 +23,21 @@ func (a ErrAdaptor) Error() string {
 // Listen() listens on a pipe, processes data, and then emits it.
 // Stop() shuts down the adaptor
 type StopStartListener interface {
+	SampleConfig() string
+	Description() string
+	Connect() error
 	Start() error
 	Listen() error
 	Stop() error
 }
 
 // Createadaptor instantiates an adaptor given the adaptor type and the Config.
-// Constructors are expected to be in the form
-//   func NewWhatever(p *pipe.Pipe, extra Config) (*Whatever, error) {}
-// and are expected to confirm to the adaptor interface
+// An Adaptor is expected to add themselves to the Adaptors map in the init() func
+//   func init() {
+//     adaptors.Add("TYPE", func(p *pipe.Pipe, path string, extra adaptors.Config) (adaptors.StopStartListener, error) {
+//     })
+//   }
+// and are expected to confirm to the Adaptor interface
```

Comment:
Majority of logic changes are here, we no longer need to use reflection to instantiate an adaptor and can pass the arguments as is to the function registered in adaptor `init`. 

The `Connect` call was added here to give adaptors a separate logical step to do any preflight checks against it's configured source.